dts: arm: nxp: i.MX RT10xx move die-temp0 from chosen to aliases

sorru94 opened the pull request:
Move `die-temp0` to the aliases section making it possible to run the `die_temp_polling` sample on NXP i.MX RT10xx boards.
Closes #86227.

fabiobaltieri commented on samples/sensor/die_temp_polling/boards/mimxrt1064_evk.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_REQUIRES_FULL_LIBC=y
```
there's already 4 overrides on this sample board directory that only has this line and none of them explains why, would it make sense to move this to the sample prj.conf? Or as a driver dependency if it depends of the driver, right now it's unclear

sorru94 replied:
Good point. 
I checked and it's not required to run this sample (at least on the RT 1064) so I removed the board configuration. 
I would not remove it from the other boards as I'm unsure about the specific dependencies.

fabiobaltieri commented on samples/sensor/die_temp_polling/boards/mimxrt1064_evk.overlay:
```diff
@@ -0,0 +1,9 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&tempmon {
+	status = "okay";
+};
```
I think this should go on the board file, don't see the point of having it as an overlay of the sample, though other boards are doing it too, may need a followup to clean this up

sorru94 replied:
What do you mean in the board file? 
I think it's appropriate for a sample to change the status of the module it uses.

fabiobaltieri replied:
I mean I'd just put this straight in `zephyrproject/zephyr/boards/nxp/mimxrt1064_evk/mimxrt1064_evk.dts`, don't see the point in laving it disabled in general and enabled only for this sample

sorru94 replied:
I'm not sure about the best practices for this type of device tree node.
Also, I don't know if enabling it by default could lead to side effects on other hardware modules.
For this PR, I would prefer to stay in line with the other boards.

fabiobaltieri replied:
yeah we don't have a solid story for that, anyway yeah fine with the current approach for this PR

fabiobaltieri reviewed (dismissed):
couple comments, though they need broader changes so they should probably be in a followup PR

sorru94 committed:
dts: arm: nxp: i.MX RT10xx move die-temp0 from chosen to aliases

Move the die-temp0 from chosen to the more appropriate aliases section.
Fixes the broken die temp polling sample for the i.MX RT10xx boards.

Signed-off-by: Simone Orru <simone.orru@secomind.com>

sorru94 committed:
samples: sensor: die_temp_polling: add i.MX RT1064 EVK support

Add the overlay file for the i.MX RT1064 EVK board.
The sample can be run on the i.MX RT1064 EVK board out of the box.

Signed-off-by: Simone Orru <simone.orru@secomind.com>
